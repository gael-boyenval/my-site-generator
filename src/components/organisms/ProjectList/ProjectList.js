import React from 'react';
import PropTypes from 'prop-types';

import Container from '../../layout/Container';
import { Margin, Padding } from '../../layout/Spaces';

import { SubTitle } from '../../atoms/Typography';

import ProjectItem from '../../molecules/ProjectItem';

const ProjectList = ({ projects }) => (
  <Padding vertical="wide">
    <Container>
      <Margin bottom="xl">
        <SubTitle>Selection de projets :</SubTitle>
      </Margin>
    </Container>

    {projects.map((project, i) => (
      <ProjectItem
        key={`${project.node.frontmatter.period}${project.node.frontmatter.client}`}
        num={`0${i + 1}`}
        {...project.node.frontmatter}
        {...project.node.fields}
      />
    ))}
  </Padding>
);

ProjectList.propTypes = {
  projects: PropTypes.arrayOf(
    PropTypes.shape({
      node: PropTypes.shape({
        frontmatter: PropTypes.shape({
          mission: PropTypes.string.isRequired,
          myRole: PropTypes.string.isRequired,
          client: PropTypes.string.isRequired,
          period: PropTypes.string.isRequired,
          brandColor: PropTypes.string.isRequired,
        }).isRequired,
        fields: PropTypes.shape({
          slug: PropTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    }),
  ).isRequired,
};

export default ProjectList;
