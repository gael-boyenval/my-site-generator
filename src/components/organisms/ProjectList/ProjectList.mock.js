export default [
  {
    node: {
      frontmatter: {
        mission: 'Refonte du site mobile',
        myRole: 'Dev Front',
        client: 'Le Monde',
        period: '2016 - 2018',
        brandColor: '#153853',
      },
      fields: {
        slug: '/projects/toto',
      },
    },
  },
  {
    node: {
      frontmatter: {
        mission: 'Conception dâ€™un back-office',
        myRole: 'UI/UX & Dev front',
        client: 'BNP Parisbas',
        period: '2016',
        brandColor: '#248d59',
      },
      fields: {
        slug: '/projects/toto',
      },
    },
  },
];
