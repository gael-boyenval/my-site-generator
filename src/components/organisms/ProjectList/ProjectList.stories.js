/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';
import ProjectList from './ProjectList';
import Mock from './ProjectList.mock';

storiesOf('04 - Organisms', module)
  .addDecorator(withKnobs)
  .add('ProjectList', withInfo()(() => <ProjectList projects={Mock} />));
