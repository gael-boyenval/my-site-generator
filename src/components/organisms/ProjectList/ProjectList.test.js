/* eslint-disable no-undef */
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';

import ProjectList from './ProjectList';
import mock from './ProjectList.mock';

const shallowWithRouter = (node) =>
  shallow(<Router initialEntries={[{ pathname: '/', key: 'testKey' }]}>{node}</Router>);

describe('Component ProjectList', () => {
  it('should be render without crash', () => {
    const component = shallowWithRouter(<ProjectList projects={mock} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
