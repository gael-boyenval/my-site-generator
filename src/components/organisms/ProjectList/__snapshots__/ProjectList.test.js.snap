// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProjectList should be render without crash 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProjectList
    projects={
      Array [
        Object {
          "node": Object {
            "fields": Object {
              "slug": "/projects/toto",
            },
            "frontmatter": Object {
              "brandColor": "#153853",
              "client": "Le Monde",
              "mission": "Refonte du site mobile",
              "myRole": "Dev Front",
              "period": "2016 - 2018",
            },
          },
        },
        Object {
          "node": Object {
            "fields": Object {
              "slug": "/projects/toto",
            },
            "frontmatter": Object {
              "brandColor": "#248d59",
              "client": "BNP Parisbas",
              "mission": "Conception dâ€™un back-office",
              "myRole": "UI/UX & Dev front",
              "period": "2016",
            },
          },
        },
      ]
    }
  />
</Router>
`;
